### PR TITLE
chore: test hyper on CI

### PR DESCRIPTION
**Pull request by taiki-e**
Split from #4390.

~~As said in #4392, this fails until https://github.com/tokio-rs/tokio/commit/cc8ad367a0e5d8536f8be58fe560bfdea1a976a5 (#4377) is reverted (or until the regression introduced by that commit is fixed).~~
UPD: https://github.com/tokio-rs/tokio/commit/cc8ad367a0e5d8536f8be58fe560bfdea1a976a5 has been reverted in #4394

### CODE REVIEW

**Review by Darksonn — APPROVED** [review]

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -355,3 +360,34 @@ jobs:
...
+          tag=$(git describe --abbrev=0 --tags)
+          git checkout "${tag}"
+          echo '[workspace]' >>Cargo.toml
+          echo '[patch.crates-io]' >>Cargo.toml
+          echo 'tokio = { path = "../tokio" }' >>Cargo.toml
+          echo 'tokio-util = { path = "../tokio-util" }' >>Cargo.toml
+          echo 'tokio-stream = { path = "../tokio-stream" }' >>Cargo.toml
+          echo 'tokio-test = { path = "../tokio-test" }' >>Cargo.toml
+          git diff
+          cargo test --features full
```
Comment (taiki-e) [inline .github/workflows/ci.yml:393]: Do we want to run benchmarks as well?
In #4390, I mentioned benchmarks, but I was actually able to reproduce both problems with only tests ([I just missed that some of hyper's benchmarks don't work on macos](https://github.com/tokio-rs/tokio/pull/4390#issuecomment-1009974709)).

Comment (Darksonn) [reply .github/workflows/ci.yml:393]: How do they work? Do they fail if something regresses, or?

Comment (carllerche) [reply .github/workflows/ci.yml:393]: cc @seanmonstar

Comment (carllerche) [reply .github/workflows/ci.yml:393]: I'm going to merge. We can always add benchmarks in a new PR.

**Pull request merged**